core/src/main/java/io/bisq/core/btc/wallet/WalletsSetup.java

```diff
@@ -300,35 +300,41 @@ private void configPeerNodesForLocalHostBitcoinNode() {
 
     private void configPeerNodes(Socks5Proxy socks5Proxy) {
         List<BitcoinNodes.BtcNode> btcNodeList = new ArrayList<>();
-        walletConfig.setMinBroadcastConnections((int) Math.floor(DEFAULT_CONNECTIONS * 0.8));
         switch (BitcoinNodes.BitcoinNodesOption.values()[preferences.getBitcoinNodesOptionOrdinal()]) {
             case CUSTOM:
-                String bitcoinNodesString = preferences.getBitcoinNodes();
-                if (bitcoinNodesString != null) {
+                final String bitcoinNodesString = preferences.getBitcoinNodes();
+                if (bitcoinNodesString != null && !bitcoinNodesString.isEmpty()) {
                     btcNodeList = Splitter.on(",")
                             .splitToList(StringUtils.deleteWhitespace(bitcoinNodesString))
                             .stream()
                             .filter(e -> !e.isEmpty())
                             .map(BitcoinNodes.BtcNode::fromFullAddress)
                             .collect(Collectors.toList());
+                    walletConfig.setMinBroadcastConnections((int) Math.ceil(btcNodeList.size() * 0.5));
+                    // If Tor is set we usually only use onion nodes, but if user provides mixed clear net and onion nodes we want to use both
+                    useAllProvidedNodes = true;
+                } else {
+                    log.warn("Custom nodes is set but no nodes are provided. We fall back to provided nodes option.");
+                    preferences.setBitcoinNodesOptionOrdinal(BitcoinNodes.BitcoinNodesOption.PROVIDED.ordinal());
+                    btcNodeList = bitcoinNodes.getProvidedBtcNodes();
+                    walletConfig.setMinBroadcastConnections(4);
                 }
                 break;
             case PUBLIC:
                 // we keep the empty list
+                walletConfig.setMinBroadcastConnections((int) Math.floor(DEFAULT_CONNECTIONS * 0.8));
                 break;
             default:
             case PROVIDED:
                 btcNodeList = bitcoinNodes.getProvidedBtcNodes();
-
                 // We require only 4 nodes instead of 7 (for 9 max connections) because our provided nodes
                 // are more reliable than random public nodes.
                 walletConfig.setMinBroadcastConnections(4);
                 break;
         }
 
-        final boolean useTorForBitcoinJ = socks5Proxy != null;
         List<PeerAddress> peerAddressList = new ArrayList<>();
-
+        boolean useTorForBitcoinJ = socks5Proxy != null;
```

Comment:
Not sure why the `final` modifier was removed here or why the line was shifted around. Leaving it off is fine for reasons mentioned elsewhere, but this change threw me off, made me wonder if I was missing something.